src/components/CreatePost/CreatePost.jsx

```diff
@@ -0,0 +1,191 @@
+import { useEffect, useState } from "react"
+import jwt_decode from "jwt-decode"
+import axios from "axios"
+import {
+    BiImageAdd,
+    AiOutlineFileGif,
+    BiPoll,
+    BsEmojiSmile,
+    IoCalendarOutline,
+    BiMap
+} from "../../assets/react-icons"
+import { useSelector, useDispatch } from "react-redux"
+import {
+  updateHomeFeed,
+  updateUserDetails
+} from "../../actions/index"
+import { 
+    useUserLogin,
+    useToast
+} from "../../context/index"
+import "./CreatePost.css"
+
+function CreatePost()
+{
+    const [ postCharCount, setPostCharCount ] = useState(280)
+    const [ newPostTextContent, setNewPostTextContent ] = useState("")
+    const { userLoggedIn } = useUserLogin()
+    const { showToast } = useToast()
+    const userDetails = useSelector(state => state.userDetailsReducer)
+    const dispatch = useDispatch()
+    const [inputTextareaPlaceholder, setInputTextareaPlaceholder] = useState("")
+    const [loggedInUserProfilePic, setLoggedInUserProfilePic] = useState("https://enztron-dev-branch.netlify.app/Icons-and-Images/Avatars/blue-illustration-avatar.svg")
+
+    useEffect(()=>{
+      let socioztronUserAuthToken = localStorage.getItem("socioztron-user-token")
+      if(socioztronUserAuthToken && JSON.stringify(userDetails)===JSON.stringify({}))
+      {
+        let loggedInUserDetails = jwt_decode(socioztronUserAuthToken)
+        dispatch(updateUserDetails({
+          loggedInUserName: loggedInUserDetails.name, 
+          loggedInUserEmail: loggedInUserDetails.email, 
+          loggedInUserProfile: loggedInUserDetails.userProfilePic
+        }))
+      }
+
+      if( JSON.stringify(userDetails)!==JSON.stringify({}) && userLoggedIn)
+      {
+        const {
+            loggedInUserName,
+            loggedInUserProfile
+        } = userDetails
+  
+        if(loggedInUserProfile==="")
+        {
+          setLoggedInUserProfilePic("https://enztron-dev-branch.netlify.app/Icons-and-Images/Avatars/blue-illustration-avatar.svg")
+        }
+        else
+        {
+          setLoggedInUserProfilePic(loggedInUserProfile)
+        }
+        setInputTextareaPlaceholder(`, ${loggedInUserName}`)
+      }
+      else
+      {
+        setInputTextareaPlaceholder("")
+      }
+    },[userDetails])
+
+    useEffect(()=>{
+      if(localStorage.getItem("socioztron-user-token")===null)
+      {
+        setInputTextareaPlaceholder("")
+      }
+    })
+
+    const createNewPost = async () => {
```

Comment:
Suggestion: you could move this function to a new file

Reply:
I think this one is fine here because I will use this function only once in the codebase and if I move it into a different file, I would need to unnecessarily pass a lot of variables that are directly accessible when placed in the same file. 
Though thanks for the suggestion, will definitely move repetitive functions to different files to avoid redundant code 😄